src/MiddlewareListener.php

```diff
@@ -115,53 +111,80 @@ public function onDispatch(MvcEvent $event)
     }
 
     /**
-     * Create a middleware pipe from the array spec given.
+     * Create a middleware pipe from the spec given.
      *
-     * @param mixed[] $middlewaresToBePiped
-     *      List of string ids for middlewares in container, instances of MiddlewareInterface
-     *      or callable middlewares
-     * @return MiddlewarePipe
+     * @param string|MiddlewareInterface|RequestHandlerInterface|PipeSpec|mixed $middlewarePipeSpec
      * @throws InvalidMiddlewareException
      */
     private function createPipeFromSpec(
-        ContainerInterface $serviceLocator,
-        ResponseInterface $responsePrototype,
-        array $middlewaresToBePiped
-    ) {
+        ContainerInterface $container,
+        $middlewarePipeSpec
+    ): RequestHandlerInterface {
+        if (is_string($middlewarePipeSpec)) {
+            $middlewarePipeSpec = $this->middlewareFromContainer($container, $middlewarePipeSpec);
+        }
+        if ($middlewarePipeSpec instanceof RequestHandlerInterface) {
+            return $middlewarePipeSpec;
+        }
+
+        if ($middlewarePipeSpec instanceof MiddlewareInterface) {
+            $middlewarePipeSpec = new PipeSpec($middlewarePipeSpec);
+        } elseif (! $middlewarePipeSpec instanceof PipeSpec) {
```

Comment:
```suggestion
        if ($middlewarePipeSpec instanceof MiddlewareInterface) {
            $middlewarePipeSpec = new PipeSpec($middlewarePipeSpec);
        }
        
         if (! $middlewarePipeSpec instanceof PipeSpec) {
```

Avoid else/elseif whenever you can, please.